### PR TITLE
matter: upmerge after test event 7.5

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Update sdk-connectedhomeip repository and Matter samples:
1. Switch to BLE random static addresses.
2. Move some attributes to external storage.
3. Enable RegulatoryConfig and LocationCapability
   attributes.
4. Update Power Source cluster enum usage.

test-sdk-nrf: PR-105
Signed-off-by: Damian Krolik <damian.krolik@nordicsemi.no>

### CODE REVIEW

File: applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf [hunk 1]
```diff
@@ -97,13 +91,13 @@ CONFIG_PWM=y
...
 CONFIG_LOG=y
+CONFIG_LOG2_MODE_DEFERRED=y
 CONFIG_LOG_BACKEND_UART=y
 CONFIG_LOG_PRINTK=y
 CONFIG_LOG_PROCESS_THREAD_STACK_SIZE=1024
 CONFIG_LOG_PROCESS_THREAD_STARTUP_DELAY_MS=4000
-CONFIG_LOG_STRDUP_BUF_COUNT=48
-CONFIG_LOG_STRDUP_MAX_STRING=100
-CONFIG_LOG_BUFFER_SIZE=2048
+CONFIG_LOG_BUFFER_SIZE=4096
```
Comment (kkasperczyk-no) [inline applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf:99]: I wonder if your fix of "unable to allocate" errors would be necessary to add here? I guess we should add this config `CONFIG_LOG_STRDUP_BUF_COUNT=100` in that case.

File: applications/matter_weather_station/src/app_task.cpp [hunk 2]
```diff
@@ -208,7 +231,7 @@ int AppTask::StartApp()
 	while (true) {
 		k_msgq_get(&sAppEventQueue, &event, K_FOREVER);
 		DispatchEvent(event);
-	}
+		}
```
Comment (kkasperczyk-no) [inline applications/matter_weather_station/src/app_task.cpp:234]: Is everything with formatting ok? I've seen similar change below and it looks for me like some bracket was removed by accident.

**Review by kkasperczyk-no — APPROVED** [review]
LGTM overall, I left only two comments to consider

File: applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf (hunk 1 above)
Comment (Damian-Nordic) [reply applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf:99]: `LOG2_MODE_DEFERRED` + log buffer size increased should be enough.

File: applications/matter_weather_station/src/app_task.cpp (hunk 2 above)
Comment (Damian-Nordic) [reply applications/matter_weather_station/src/app_task.cpp:234]: Right, I'll reformat the file.

**Review by wiba-nordic — APPROVED** [review]
Looks good on the documentation front.

**Pull request merged**